utils/sysfs/sysfs.go

```diff
@@ -98,7 +98,25 @@ func (self *realSysFs) GetBlockDeviceSize(name string) (string, error) {
 }
 
 func (self *realSysFs) GetNetworkDevices() ([]os.FileInfo, error) {
-	return ioutil.ReadDir(netDir)
+	files, err := ioutil.ReadDir(netDir)
+	if err != nil {
+		return nil, err
+	}
+
+	// Filter out non-directory & non-symlink files
+	var dirs []os.FileInfo
+	for _, f := range files {
+		if f.Mode()|os.ModeSymlink != 0 {
+			f, err = os.Stat(path.Join(netDir, f.Name()))
+			if err != nil {
+				continue
+			}
+		}
```

Comment:
hmm. os.Stat() will follow eth0 symlink and modify f with the symlinked original entry(../../devices/pci0000:00/0000:00:04.0/virtio1/net/eth0) which IS a directory.. no?